File: hw/Vaca/u3/HW30ObserverAndStrategy/HW30Strategy/src/main/java/ec/edu/espe/controller/InsertionSort.java

package ec.edu.espe.controller;

/**
 *
 * @author MSI-PULSE
 */
public class InsertionSort implements SortingStrategy {
    @Override
    public void sort(int[] numbers) {
        int n = numbers.length;
        for (int i = 1; i < n; ++i) {
            int key = numbers[i];
            int j = i - 1;

            // Mueve los elementos del arreglo que son mayores que la clave a una posición adelante de su posición actual
            while (j >= 0 && numbers[j] > key) {
                numbers[j + 1] = numbers[j];
                j = j - 1;
            }
            numbers[j + 1] = key;
        }
    }

    @Override
    public String getStrategyName() {
        return "Insertion Sort";
    }
}

File: exams/viteri/U3/FinalExam Viteri/FinalExam Viteri/utils/Insertion_Sort.py
class Insertion_Sort:
    def insertion_sort(list)
        n=10
        for int i = 1; i < n; ++i: 
            int key = list[i]
            int j = i - 1
            while j >= 0 && list[j] > key: 
                list[j + 1] = list[j]
                j = j - 1
            
            list[j + 1] = key